pkgs/development/compilers/llvm/common/default.nix

```
@@ -452,14 +452,21 @@ let
               stripLen = 1;
             })
           ]
-          ++ lib.optionals (lib.versions.major metadata.release_version == "14") [
-            # fix RuntimeDyld usage on aarch64-linux (e.g. python312Packages.numba tests)
-            (fetchpatch {
-              url = "https://github.com/llvm/llvm-project/commit/2e1b838a889f9793d4bcd5dbfe10db9796b77143.patch";
-              relative = "llvm";
-              hash = "sha256-Ot45P/iwaR4hkcM3xtLwfryQNgHI6pv6ADjv98tgdZA=";
-            })
-          ]
+          ++
+            lib.optionals
+              (
+                (lib.versionAtLeast (lib.versions.major metadata.release_version) "14")
+                && (lib.versionOlder (lib.versions.major metadata.release_version) "18")
```

```suggestion
                && (lib.versionOlder (lib.versions.major metadata.release_version) "17")
```

patch is already applied in llvm-17:
https://github.com/llvm/llvm-project/blob/6009708b4367171ccdbf4b5905cb6a803753fe18/llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldELF.cpp#L2413